content/en/docs/tasks/manage-kubernetes-objects/storage-version-migration.md

```diff
@@ -0,0 +1,302 @@
+  ```shell
+  kubectl apply -f migrate-secret.yaml
+  ```
+- Monitor migration of secrets using status. Successful migration should have
+_Succeeded_ condition set to _true_ in the status field. Get the migration resource
+as follows:
+  ```shell
+  kubectl get storageversionmigration.storagemigration.k8s.io/secrets-migration -o yaml
+  ```
+  This will produce an output similar to:
+  ```yaml
+  kind: StorageVersionMigration
+  apiVersion: storagemigration.k8s.io/v1alpha1
+  metadata:
+    name: secrets-migration
+    uid: 628f6922-a9cb-4514-b076-12d3c178967c
+    resourceVersion: '90'
+    creationTimestamp: '2024-03-12T20:29:45Z'
+  spec:
+    resource:
+      group: ""
+      version: v1
+      resource: secrets
+  status:
+    conditions:
+    - type: Running
+      status: 'False'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationInProgress
+    - type: Succeeded
+      status: 'True'
+      lastUpdateTime: '2024-03-12T20:29:46Z'
+      reason: StorageVersionMigrationSucceeded
+    resourceVersion: '84'
+  ```
+- [Verify](/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted)
+  the stored secret is now prefixed with `k8s:enc:aescbc:v1:key2`.
+
+## Update the preferred storage schema of the resource while moving from _v1_ to _v2_
+Consider the scenario where a CustomResourceDefinition (CRD) is created to serve
+Custom Resources (CRs) and is set as the preferred storage schema. When it's time
+to introduce v2 of the CRD, it can be added for serving only with a conversion
+webhook. This enables a smoother transition where users can create CRs using
+either the v1 or v2 schema, with the webhook in place to perform the necessary
+schema conversion between them. Before setting v2 as the preferred storage schema
+version, it's important to ensure that all existing CRs stored as v1 are migrated to v2.
+This migration can be achieved through _Storage Version Migration_ to migrate all CRs from v1 to v2.
+
+- Create the CRD resource file with the name `test-crd.yaml` as follows:
+  ```yaml
+  apiVersion: apiextensions.k8s.io/v1
+  kind: CustomResourceDefinition
+  metadata:
+    name: testcrds.stable.example.com
+  spec:
+    group: stable.example.com
+    names:
+      plural: testcrds
+      singular: testcrd
+      kind: testcrd
+      listKind: testcrdList
+    scope: Namespaced
+    versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            hostPort:
+              type: string
+    conversion:
+      strategy: Webhook
+      webhook:
+        clientConfig:
+          url: https://127.0.0.1:9443/crdconvert
+          caBundle: <CABundle info>
+      conversionReviewVersions:
+      - v1
+      - v2
+  ```
+  Create CRD using kubectl
+  ```shell
+  kubectl apply -f test-crd.yaml
+  ```
+- Create Custom Resource (CR) resource file with name `cr1.yaml` as follows:
```

Comment:
Maybe:
```suggestion
- Create a manifest for an example testcrd. Name the manifest `cr1.yaml` and use these contents:
```

IMO, it's even better to use an plausible but obviously example API kind like FruitClaim or RoomTemperature or SelfieRequest.